lib/wallet/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -728,22 +749,25 @@ mkUnsignedTransaction networkId stakeCred ctx selection = do
                             inpsScripts Nothing refScriptM
                 _ ->
                     constructUnsignedTx
-                        networkId (md, []) ttl wdrl
+                        networkId (md, votingCerts) ttl wdrl
                         selection delta assetsToBeMinted assetsToBeBurned
                         inpsScripts
                     Nothing refScriptM
         Just action -> do
-            let certs = case stakeCred of
+            let delegCerts = case stakeCred of
                     Left xpub ->
-                        certificateFromDelegationAction era (Left xpub) action
+                        certificateFromDelegationAction era (Left xpub)
+                        depositM action
                     Right (Just script) ->
-                        certificateFromDelegationAction era (Right script) action
+                        certificateFromDelegationAction era (Right script)
+                        depositM action
                     Right Nothing ->
                         error $ unwords
                             [ "stakeCred in mkUnsignedTransaction must be"
                             , "either xpub or script when there is delegation"
                             , "action"
                             ]
+            let certs = L.nub $ votingCerts <> delegCerts
```

Comment:
👍  I have checked. In Conway, the rule handling the certificates is called `CERTS`. As far as I can tell, it goes left to right, so `nub` is appropriate here.